tests/integration/functional/steps/test_model_config.py

```diff
@@ -807,3 +841,110 @@ def _inner_pipeline():
         model = client.get_model(model_name_or_id="step")
         assert len(model.versions) == 1
         assert len(model.versions[0].artifact_object_ids) == 0
+
+
+@pytest.mark.skipif(
+    platform.system() == "Windows",
+    reason="Subprocesses on Windows are hardly manageable.",
+)
+@pytest.mark.parametrize(
+    "version",
+    ("test running version", ""),
+    ids=["custom_running_name", "default_running_name"],
+)
+def test_that_two_pipelines_cannot_run_at_the_same_time_requesting_new_version_and_with_recovery(
+    version,
+):
+    """Test that if second pipeline for same new version is started in parallel - it will fail."""
+    with model_killer():
+        long_runtime = 120
+        long_run_name = f"long_{uuid4()}"
+        # start a long running pipeline with a request for a new version
+        long_process = subprocess.Popen(
```

Comment:
Very cool you decided to test this via actual parallelism 👍 👏 

Only thing I was wondering, is there a specific reason why you use `subprocess` over `multiprocessing`?

In the latter case you wouldn't need to leave Python, e.g. this should work:

```python
from multiprocessing import Process

p1 = Process(target=lambda: pipeline_1())
p1.start()
p2 = Process(target=lambda: pipeline_2())
p2.start()
p1.join()
p2.join()
```

Reply:
I started with this, but it was not working as expected in the CI of Github, not sure why, so I had to switch to `subprocess` which is a more OS-natural approach.

Reply:
But I tried with `Pool` actually, not sure if `Process` would be any better, but since I'll do some rework anyway today, I'll give a shot to this proposal as well.

Reply:
I restored what happened back then - in CI it will start throwing `Unable to determine source root because the main module does not have an associated file. This could be because you're running in an interactive Python environment.`, thus I had to switch to a subprocess calling script.